File: frontend/blog-web/src/api/category.ts
import request from '@/utils/request'
import type { ApiResult } from '@/types/api'

export interface CategoryVO {
  id: number
  name: string
  slug: string
  description: string
  sort: number
  status: number
}

export function getAllCategories() {
  return request.get<ApiResult<CategoryVO[]>>('/blog/categories')
}

File: services/blog-service/src/main/java/com/platform/blog/domain/entity/Category.java
package com.platform.blog.domain.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.platform.common.core.domain.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 分类实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("biz_category")
public class Category extends BaseEntity {

    /**
     * 分类名称
     */
    private String name;

    /**
     * 分类别名（URL友好）
     */
    private String slug;

    /**
     * 描述
     */
    private String description;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 状态: 1-启用, 0-禁用
     */
    private Integer status;
}
